core/src/validator.rs

```diff
@@ -1111,19 +1113,38 @@ impl Validator {
         let entry_notification_sender = entry_notifier_service
             .as_ref()
             .map(|service| service.sender_cloned());
-        let turbine_quic_connection_cache = Arc::new(
-            solana_quic_client::new_quic_connection_cache(
-                "connection_cache_tvu_quic",
-                &identity_keypair,
-                node.info
-                    .tvu(Protocol::QUIC)
-                    .expect("Operator must spin up node with valid TVU address")
-                    .ip(),
-                &staked_nodes,
-                TURBINE_QUIC_CONNECTION_POOL_SIZE,
-            )
-            .unwrap(),
-        );
+
+        // test-validator crate may start the validator in a tokio runtime
+        // context which forces us to use the same runtime because a nested
+        // runtime will cause panic at drop.
+        let current_runtime_handle = tokio::runtime::Handle::try_current();
+        let turbine_quic_endpoint_runtime = current_runtime_handle.is_err().then(|| {
+            tokio::runtime::Builder::new_multi_thread()
+                .enable_all()
+                .thread_name("solTurbineQuic")
+                .build()
+                .unwrap()
+        });
+        let (turbine_quic_endpoint_sender, turbine_quic_endpoint_receiver) = unbounded();
+        let (
+            turbine_quic_endpoint,
+            turbine_quic_endpoint_sender,
+            turbine_quic_endpoint_join_handle,
+        ) = solana_turbine::quic_endpoint::new_quic_endpoint(
+            turbine_quic_endpoint_runtime
+                .as_ref()
+                .map(tokio::runtime::Runtime::handle)
+                .unwrap_or_else(|| current_runtime_handle.as_ref().unwrap()),
+            &identity_keypair,
+            node.sockets.tvu_quic,
+            node.info
+                .tvu(Protocol::QUIC)
+                .expect("Operator must spin up node with valid TVU address")
```

Comment:
nit to be clearer that it's a different address
```suggestion
                .expect("Operator must spin up node with valid QUIC TVU address")
```